autometa/common/external/diamond.py

```diff
@@ -29,146 +29,301 @@
-    if retcode:
-        raise OSError(f'DiamondFailed:\nArgs:{proc.args}\nReturnCode:{proc.returncode}')
+
+def makedatabase(fasta, database, nproc=mp.cpu_count()):
+    """
+    Creates a database against which the query sequence would be blasted
+
+    Parameters
+    ----------
+    fasta : str
+        Path to fasta file whose database needs to be made
+        e.g. '<path/to/fasta/file>'
+    database : str
+        Path to the output  nr database file
+        e.g. '<path/to/database/file>'
+    nproc : int, optional
+        Number of processors to be used. By default uses all the processors of the system 
+
+    Returns
+    -------
+    str
+        Path to Diamond database
+
+    Raises
+    ------
+    CalledProcessError
+        Failed to create Diamond database
+    """
+    cmd = f"diamond makedb --in {fasta} --db {database} -p {nproc}"
+    logger.debug(f"{cmd}")
+    proc = subprocess.run(
+        cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, text=True
+    )
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError as err:
+        raise err
     return database
 
-def blast(fasta, database, outfpath, blast_type='blastp', evalue=float('1e-5'),
-    maxtargetseqs=200, cpus=0, tmpdir=os.curdir, force=False, verbose=False):
-    """Performs diamond blastp search using fasta against diamond formatted database
+
+def blast(
+    fasta,
+    database,
+    outfpath,
+    blast_type="blastp",
+    evalue=float("1e-5"),
+    maxtargetseqs=200,
+    nproc=mp.cpu_count(),
+    tmpdir=os.curdir,
+    force=False,
+    verbose=False,
+):
+    """
+    Performs diamond blastp search using query sequence against diamond formatted database
 
     Parameters
     ----------
     fasta : str
-        </path/to/fasta/file>. May be amino acid or nucleotide sequences
+        Path to fasta file having the query sequence. May be amino acid or nucleotide sequences
+        '</path/to/fasta/file>' 
     database : str
-        </path/to/diamond/formatted/database>.
+        Path to diamond formatted database
+        '</path/to/diamond/formatted/database>'
     outfpath : str
-        </path/to/output/file>.
-    blast_type : str
-        blastp if fasta consists of amino acids or blastx if nucleotides
-    evalue : float
-        cutoff e-value to count hit as significant (the default is float('1e-5')).
-    maxtargetseqs : int
-        max number of target sequences to retrieve per query by diamond (the default is 200).
-    cpus : int
-        number of cpus to use (the default is 1).
-    tmpdir : type
-        </path/to/temporary/directory> (the default is os.curdir).
-    force : boolean
-        overwrite existing diamond results `force` (the default is False).
-    verbose : boolean
-        log progress to terminal `verbose` (the default is False).
+        Path to output file
+        '</path/to/output/file>'
```

Comment:
I'm not sure having both of these lines is helpful.

```suggestion

```

Reply:
I understand that this is a biit redundant but I was thinking we should keep this to maintain consistency. Let me know what you think